src/borg/cache.py

```diff
@@ -15,15 +15,18 @@
 from .helpers import Location
 from .helpers import Error
 from .helpers import get_cache_dir
-from .helpers import decode_dict, int_to_bigint, bigint_to_int, bin_to_hex
+from .helpers import safe_encode, decode_dict, int_to_bigint, bigint_to_int, bin_to_hex
 from .helpers import format_file_size
 from .helpers import yes
 from .item import Item, ArchiveItem
 from .key import PlaintextKey
 from .locking import Lock
 from .platform import SaveFile
+from .platform import fstype
 from .remote import cache_if_remote
 
+FS_WITH_STABLE_INODES = {'extfs', 'btrfs', 'xfs', 'zfs', }
```

Comment:
hmm, move this to platform code?

guess it depends on the platform's implementation of a filesystem whether it has stable inodes?